sirepo/pkcli/service.py

```diff
@@ -13,63 +13,75 @@
 from pykern import pkio
 from pykern import pkjinja
 from pykern import pksubprocess
-from pykern.pkdebug import pkdc, pkdexc, pkdp
+from pykern.pkdebug import pkdc, pkdexc, pkdp, pkdlog
+import errno
+import os
 import py
 import re
+import signal
 import sirepo.srdb
 import socket
+import subprocess
+import sys
 
 
-def celery():
-    """Start celery"""
-    assert pkconfig.channel_in('dev')
-    import celery.bin.celery
-    import sirepo.celery_tasks
-    run_dir = _run_dir().join('celery').ensure(dir=True)
-    with pkio.save_chdir(run_dir):
-        celery.bin.celery.main(argv=[
-            'celery',
-            'worker',
-            '--app=sirepo.celery_tasks',
-            '--no-color',
-            '-Ofair',
-            '--queue=' + ','.join(sirepo.celery_tasks.QUEUE_NAMES),
-        ])
-
-
-def flower():
-    """Start flower"""
-    from flower import command
-    assert pkconfig.channel_in('dev')
-    run_dir = _run_dir().join('flower').ensure(dir=True)
-    with pkio.save_chdir(run_dir):
-        command.FlowerCommand().execute_from_commandline([
-            'flower',
-            '--address=' + cfg.ip,
-            '--app=sirepo.celery_tasks',
-            '--no-color',
-            '--persistent',
-        ])
+def flask():
+    from sirepo import server
+
+    use_reloader = pkconfig.channel_in('dev')
+    app = server.init(use_reloader=use_reloader)
+    # avoid WARNING: Do not use the development server in a production environment.
+    app.env = 'development'
+    app.run(
+        host=cfg.ip,
+        port=cfg.port,
+        threaded=True,
+        use_reloader=use_reloader,
+    )
 
 
 def http():
-    """Starts Flask server in http mode.
+    """Starts the Flask server and job_supervisor.
 
     Used for development only.
     """
-    from sirepo import server
-
-    with pkio.save_chdir(_run_dir()):
-        use_reloader = pkconfig.channel_in('dev')
-        app = server.init(use_reloader=use_reloader)
-        # avoid WARNING: Do not use the development server in a production environment.
-        app.env = 'development'
-        app.run(
-            host=cfg.ip,
-            port=cfg.port,
-            threaded=True,
-            use_reloader=use_reloader,
+    def _env():
+        e = os.environ
```

Comment:
This needs to be a copy, because you don't want to modify this process' environ. Don't need update for one value so:

```py
e = PKDict(os.environ)
e. SIREPO_JOB_DRIVER_MODULES = 'local'
return e
```

And this could be done once and used by _start. It doesn't need to be a routine, because it is invariant with respect to the two processes. Just need to define `e` before `_start` is defined.